samples/logging_s3/pipeline_log_to_s3_by_fluentd_recommend/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM fluent/fluentd:v1.11-1
+MAINTAINER fenglixa <fenglixa@cn.ibm.com>
+
+USER root
+
+RUN apk add --no-cache --update --virtual .build-deps \
+        sudo build-base ruby-dev \
+        && sudo gem install fluent-plugin-s3 \
+        && sudo gem install fluent-plugin-kubernetes_metadata_filter \
+        && sudo gem install fluent-plugin-rewrite-tag-filter \
+        && sudo gem sources --clear-all \
+        && apk del .build-deps \
+        && rm -rf /home/fluent/.gem/ruby/2.5.0/cache/*.gem
+
+COPY fluent.conf /fluentd/etc/
```

Comment:
Thanks @fenglixa, for this fluent.conf, I'm more thinking of loading this file from a configmap like this https://github.com/knative/serving/blob/master/config/monitoring/logging/elasticsearch/100-fluentd-configmap.yaml#L27-L90

Otherwise, it looks good to me.